core/main/src/firebolt/handlers/advertising_rpc.rs

```diff
@@ -330,76 +338,107 @@ impl AdvertisingServer for AdvertisingImpl {
             .await
             .unwrap_or(false);
 
-        let payload = AdvertisingRequest::GetAdInitObject(AdInitObjectRequestParams {
-            privacy_data: privacy_rpc::get_allow_app_content_ad_targeting_settings(
-                &self.state,
-                None,
-                &app_id,
-            )
-            .await,
-            environment: config.options.environment.to_string(),
-            durable_app_id: app_id,
-            app_version: "".to_string(),
-            distributor_app_id: distributor_experience_id,
-            device_ad_attributes: HashMap::new(),
-            coppa: config.options.coppa.unwrap_or(false),
-            authentication_entity: config.options.authentication_entity.unwrap_or_default(),
+        let ad_opt_out = PrivacyImpl::get_allow_app_content_ad_targeting(&self.state).await;
+
+        let mut privacy_data = privacy_rpc::get_allow_app_content_ad_targeting_settings(
+            &self.state,
+            None,
+            &durable_app_id,
+        )
+        .await;
+
+        // RPPL-1413: Switch to GetAdRouting API instead of GetAdInitObject on APS: GetAdInitObject would modify the passed-in
+        // privacy data to include a hard-coded pdt value. Temporarily hard-coding on ripple-side until we deteremine what
+        // it's for and if it's still needed.
+        privacy_data.insert("pdt".into(), "gdp:v1".into());
+
+        let coppa = match config.options.coppa {
+            Some(c) => c as u32,
+            None => 0,
+        };
+
+        let advertising_request = AdvertisingRequest::GetAdConfig(AdConfigRequestParams {
+            privacy_data: privacy_data.clone(),
+            durable_app_id: durable_app_id.clone(),
             dist_session: session
+                .clone()
                 .ok_or_else(|| Error::Custom(String::from("no session available")))?,
-            scope: HashMap::new(),
+            environment: environment.to_string(),
+            scope: get_scope_option_map(&None),
```

Comment:
Currently the firebolt `advertising.config` API doesn't include a `scope` parameter like the `advertising.id` API does, it seems like the spec should include it since `advertising.config` now gets the xifa directly. Was this a miss when `advertising.config` called `GetAdInitObject`?

Reply:
advertising.config will only return the default root scope, we don't support using advertising.config to get other scoped IFAs